packages/crypto/src/crypto.ts

```diff
@@ -0,0 +1,498 @@
+import { btoa, atob } from "react-native-quick-base64";
+import { p256 } from "@noble/curves/p256";
+import * as hkdf from "@noble/hashes/hkdf";
+import { sha256 } from "@noble/hashes/sha256";
+import { gcm } from "@noble/ciphers/aes";
+import * as bs58check from "bs58check";
+
+import {
+  AES_KEY_INFO,
+  HPKE_VERSION,
+  IV_INFO,
+  LABEL_EAE_PRK,
+  LABEL_SECRET,
+  LABEL_SHARED_SECRET,
+  SUITE_ID_1,
+  SUITE_ID_2,
+} from "./constants";
+
+interface HPKEDecyptParams {
+  ciphertextBuf: Uint8Array;
+  encappedKeyBuf: Uint8Array;
+  receiverPriv: string;
+}
+
+interface KeyPair {
+  privateKey: string;
+  publicKey: Uint8Array | string;
+  publicKeyUncompressed?: string;
+}
+
+/**
+ * Get PublicKey function
+ * Derives public key from Uint8Array private key
+ *
+ * @param {Uint8Array} privateKey - The Uint8Array representation of a private key
+ * @param {boolean} isCompressed - true by default, specifies whether to return a compressed or uncompressed public key
+ * @returns {Promise<Uint8Array>} - The public key in Uin8Array representation.
+ */
+
+export const getPublicKey = (
+  privateKey: Uint8Array,
+  isCompressed: boolean = true
+): Uint8Array => {
+  return p256.getPublicKey(privateKey, isCompressed);
+};
+/**
+ * HPKE Decrypt Function
+ * Decrypts data using Hybrid Public Key Encryption (HPKE) approach.
```

Comment:
this is more than an approach, it's a standard! Let's link https://datatracker.ietf.org/doc/rfc9180/